cmd/participant/landing.go

```diff
@@ -107,14 +107,17 @@ func (c *Client) TryContribute() error {
 		return err
 	}
 
+	if resp.StatusCode >= 200 && resp.StatusCode < 300 {
+		// Success!
+		return nil
+	}
+
 	var errorUnm ErrorStruct
-	if err := json.Unmarshal(responseData, &errorUnm); err == nil {
-		fmt.Printf("Error: %v", errorUnm)
-		return errors.New(errorUnm.Error)
+	if err := json.Unmarshal(responseData, &errorUnm); err != nil {
+		return fmt.Errorf("error unmarshaling error data \"%v\": %w", string(responseData), err)
```

Comment:
Fun bug: Go won't error if a json field is missing. So when the sequencer let me through, I got as a response `Error attempting to contribute: :  ` because it successfully unmarshalled the response as an error despite neither the code nor error fields existing.